app/src/main/java/net/programmierecke/radiodroid2/FragmentSettings.java

```diff
@@ -36,83 +45,155 @@
-                }
+    public Fragment getCallbackFragment() {
+        return this;
+    }
 
-                return false;
-            }
-        });
+    public static FragmentSettings openNewSettingsSubFragment(ActivityMain activity, String key) {
+        FragmentSettings f = new FragmentSettings();
+        Bundle args = new Bundle();
+        args.putString(PreferenceFragmentCompat.ARG_PREFERENCE_ROOT, key);
+        f.setArguments(args);
+        FragmentTransaction fragmentTransaction = activity.getSupportFragmentManager().beginTransaction();
+        fragmentTransaction.replace(R.id.containerView, f).addToBackStack(String.valueOf(FRAGMENT_FROM_BACKSTACK)).commit();
+        return f;
+    }
 
-        findPreference("settings_proxy").setOnPreferenceClickListener(new OnPreferenceClickListener() {
-            @Override
-            public boolean onPreferenceClick(Preference preference) {
-                ProxySettingsDialog proxySettingsDialog = new ProxySettingsDialog();
-                proxySettingsDialog.setCancelable(true);
-                proxySettingsDialog.show(getFragmentManager(), "");
-                return false;
-            }
-        });
+    @Override
+    public boolean onPreferenceStartScreen(PreferenceFragmentCompat preferenceFragmentCompat,
+                                           PreferenceScreen preferenceScreen) {
+        openNewSettingsSubFragment((ActivityMain) getActivity(), preferenceScreen.getKey());
+        return true;
+    }
 
-        findPreference("mpd_servers_viewer").setOnPreferenceClickListener(new Preference.OnPreferenceClickListener() {
-            @Override
-            public boolean onPreferenceClick(Preference preference) {
-                showAllMPDServers();
-                return false;
-            }
-        });
+    private boolean isToplevel() {
+        return getPreferenceScreen() == null || getPreferenceScreen().getKey().equals("pref_toplevel");
+    }
 
-        findPreference("show_statistics").setOnPreferenceClickListener(new Preference.OnPreferenceClickListener() {
-            @Override
-            public boolean onPreferenceClick(Preference preference) {
-                ((ActivityMain)getActivity()).getToolbar().setTitle(R.string.settings_statistics);
-                FragmentServerInfo f = new FragmentServerInfo();
-                FragmentTransaction fragmentTransaction = getFragmentManager().beginTransaction();
-                fragmentTransaction.replace(R.id.containerView, f).addToBackStack(String.valueOf(FRAGMENT_FROM_BACKSTACK)).commit();
-                return false;
-            }
-        });
+    private void refreshToplevelIcons() {
+        findPreference("shareapp_package").setSummary(getPreferenceManager().getSharedPreferences().getString("shareapp_package", ""));
+        findPreference("pref_category_ui").setIcon(Utils.IconicsIcon(getActivity(), "cmd_monitor"));
```

Comment:
Better use enum like `CommunityMaterial.Icon2.cmd_monitor`and not hardcoded strings.

Reply:
indeed, will change...